Project-Code/src/index.js

```diff
@@ -103,10 +103,12 @@ app.post('/login', async (req,res) => {
       bcrypt.compare(req.body.password, data.password)
         .then((match) => {
           if (match) {
+            user.id = data.user_id;
             user.username = data.username;
-            user.password = data.password;
-    
-            req.session.user = user;
+            user.password = data.password;  
+            user.booksRead = data.books_read;
```

Comment:
Here the user key we are trying to access doesn't exist because of a minor spelling difference.